File: concat_markets.py
#!/usr/bin/env python3

import json
import os
import glob
from pathlib import Path

def main():
    # Define paths
    core_data_dir = "core-data"
    output_file = "fixtures/markets.json"
    
    # Ensure fixtures directory exists
    os.makedirs("fixtures", exist_ok=True)
    
    # Find all market files
    market_files = glob.glob(os.path.join(core_data_dir, "*-markets.json"))
    
    all_markets = []
    
    for market_file in sorted(market_files):
        # Extract league name from filename (e.g., "ENG1-markets.json" -> "ENG1")
        filename = os.path.basename(market_file)
        league = filename.split("-markets.json")[0]
        
        print(f"Processing {filename} (league: {league})")
        
        try:
            # Load the market file
            with open(market_file, 'r') as f:
                markets = json.load(f)
            
            # Add league field to each market
            for market in markets:
                market["league"] = league
                all_markets.append(market)
                
            print(f"  Added {len(markets)} markets for {league}")
            
        except Exception as e:
            print(f"  Error processing {market_file}: {e}")
            continue
    
    # Save concatenated results
    print(f"\nSaving {len(all_markets)} total markets to {output_file}")
    
    with open(output_file, 'w') as f:
        json.dump(all_markets, f, indent=2)
    
    print("✓ Markets concatenation complete!")
    
    # Show summary by league
    league_counts = {}
    for market in all_markets:
        league = market["league"]
        league_counts[league] = league_counts.get(league, 0) + 1
    
    print("\nSummary by league:")
    for league, count in sorted(league_counts.items()):
        print(f"  {league}: {count} markets")

if __name__ == "__main__":
    main()

File: concat_markets.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"path/filepath"
	"sort"
	"strings"
)

func main() {
	coreDataDir := "core-data"
	outputFile := "fixtures/markets.json"

	if err := os.MkdirAll("fixtures", 0755); err != nil {
		fmt.Printf("Error creating fixtures directory: %v\n", err)
		os.Exit(1)
	}

	pattern := filepath.Join(coreDataDir, "*-markets.json")
	marketFiles, err := filepath.Glob(pattern)
	if err != nil {
		fmt.Printf("Error finding market files: %v\n", err)
		os.Exit(1)
	}

	sort.Strings(marketFiles)

	var allMarkets []map[string]interface{}

	for _, marketFile := range marketFiles {
		filename := filepath.Base(marketFile)
		league := strings.TrimSuffix(filename, "-markets.json")

		fmt.Printf("Processing %s (league: %s)\n", filename, league)

		data, err := os.ReadFile(marketFile)
		if err != nil {
			fmt.Printf("  Error reading %s: %v\n", marketFile, err)
			continue
		}

		var markets []map[string]interface{}
		if err := json.Unmarshal(data, &markets); err != nil {
			fmt.Printf("  Error parsing %s: %v\n", marketFile, err)
			continue
		}

		for i := range markets {
			markets[i]["league"] = league
			allMarkets = append(allMarkets, markets[i])
		}

		fmt.Printf("  Added %d markets for %s\n", len(markets), league)
	}

	fmt.Printf("\nSaving %d total markets to %s\n", len(allMarkets), outputFile)

	outputData, err := json.MarshalIndent(allMarkets, "", "  ")
	if err != nil {
		fmt.Printf("Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}

	if err := os.WriteFile(outputFile, outputData, 0644); err != nil {
		fmt.Printf("Error writing output file: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("✓ Markets concatenation complete!")

	leagueCounts := make(map[string]int)
	for _, market := range allMarkets {
		if league, ok := market["league"].(string); ok {
			leagueCounts[league]++
		}
	}

	fmt.Println("\nSummary by league:")
	var leagues []string
	for league := range leagueCounts {
		leagues = append(leagues, league)
	}
	sort.Strings(leagues)

	for _, league := range leagues {
		fmt.Printf("  %s: %d markets\n", league, leagueCounts[league])
	}
}